src/devtools/client/debugger/src/utils/sourceVisualizations.ts

```diff
@@ -170,9 +180,10 @@ function getAlternateSourceIdForPosition(
   client: ReplayClientInterface,
   source: SourceDetails,
   position: CursorPosition,
-  sourcesById: Dictionary<SourceDetails>
+  sourcesById: Dictionary<SourceDetails>,
+  sourceLocationRange: SourceLocationRange | null
 ) {
-  const lineLocations = getBreakpointPositionsSuspense(client, source.id, null);
+  const lineLocations = getBreakpointPositionsSuspense(client, source.id, sourceLocationRange);
```

Comment:
Requesting all breakpoint positions for the source (i.e. without a range) certainly wasn't optimal but I don't think we should use the currently *visible* source range here since we're interested in finding a breakable position after the currently *selected* position and that position may not be in the visible range. This has two downsides:
- if the currently selected position is below the currently visible range the SourcemapToggle will always say "The currently selected location is not mapped" even if that is not the case
- we rerender the SourcemapToggle every time the currently visible range changes

I think we should use a range starting with the currently selected line and ending maybe 10 lines below that.

Reply:
> I don't think we should use the currently visible source range here since we're interested in finding a breakable position after the currently selected position and that position may not be in the visible range

I don't understand what you mean by this.

> if the currently selected position is below the currently visible range the SourcemapToggle will always say "The currently selected location is not mapped" even if that is not the case...I think we should use a range starting with the currently selected line and ending maybe 10 lines below that.

Why would 10 lines help? If there's a concern that something could be selected but not within the view port, then wouldn't that still be possible?

Btw, I don't think we're ever passing the visible range param to these functions. (It's optional.)

PS. You don't need to keep marking "_requested changes_" 😄  This is a large refactor. I'm not going to merge it until everyone is onboard.

Reply:
When I say: _I don't understand what you mean by this_

What I mean is: Can you describe a specific scenario that would cause a problem here?

Reply:
> I don't understand what you mean by this.

We look for the first entry in `lineLocations` whose `line` is greater or equal to `position.line`. If the range for which we requested the breakable positions doesn't include `position.line` then we won't find any and hence we can't get a `MappedLocation` (because that only works for breakable positions) and hence we can't find an alternate source.

> Why would 10 lines help? 

It's not about the 10 lines, it's about the range including `position.line`. Since we're only interested in breakable positions *after* that line, we don't need any earlier lines in the range but the next breakable position may be at a line after `position.line`, so the range needs to include some lines after `position.line`. We could use a range starting with `position.line` and ending with the last line of the source but that could become a perf problem (just like not using a range at all). I think it's reasonable if there are no breakable positions on the first 10 lines after `position.line` to say we can't find an alternate source.

> Can you describe a specific scenario that would cause a problem here?

Select a position in a bundle that you're not currently paused in and that can be mapped to an original source. As long as the selected position is visible, the SourcemapToggle will let you switch to that original source. But as soon as you scroll the source up so that the selected position is not visible anymore, the SourcemapToggle gets disabled and says "The currently selected location is not mapped". Interested in a replay?

Reply:
> Since we're only interested in breakable positions _after_ that line

I don't understand this part. I thought we were specifically looking for other positions _on that same line_. Why would we look for ones on _after_ the specified line?

Reply:
tl;dr I'm going to remove the visible lines range filtering from this Suspense cache. Looking at production, we're actually fetching this data _multiple times_ currently– once without any visible lines filtered, and once with them filtered (or once per chunk of visible lines).

So removing the filtering altogether will make things _better_. (We'll only fetch once and then read from memory.) It will also side-step this issue.